packages/playback-core/src/index.ts

```diff
@@ -23,174 +30,41 @@ export const generatePlayerInitTime = () => {
   if (!playbackId) return undefined;
   const [idPart, queryPart = ''] = toPlaybackIdParts(playbackId);
   return `https://stream.${domain}/${idPart}.m3u8${queryPart}`;
 };
 
-export const inferMimeTypeFromURL = (url: string) => {
-  let pathname = '';
-  try {
-    pathname = new URL(url).pathname;
-  } catch (e) {
-    console.error('invalid url');
-  }
-
-  const extDelimIdx = pathname.lastIndexOf('.');
-  if (extDelimIdx < 0) return '';
-
-  const ext = pathname.slice(extDelimIdx + 1);
-  const upperExt = ext.toUpperCase();
-
-  return isKeyOf(upperExt, ExtensionMimeTypeMap) ? ExtensionMimeTypeMap[upperExt] : '';
-};
-
-export const getType = (props: Partial<Pick<MuxMediaProps, 'type' | 'src'>>) => {
-  const type = props.type;
-
-  if (type) {
-    const upperType = type.toUpperCase();
-
-    return isKeyOf(upperType, MimeTypeShorthandMap) ? MimeTypeShorthandMap[upperType] : type;
-  }
-
-  const { src } = props;
-
-  if (!src) return '';
-
-  return inferMimeTypeFromURL(src);
+export const getError = (mediaEl: HTMLMediaElement) => {
+  return muxMediaState.get(mediaEl)?.error;
 };
 
-export const getStreamTypeConfig = (streamType?: ValueOf<StreamTypes>) => {
-  // for regular live videos, set backBufferLength to 8
-  if ([StreamTypes.LIVE, StreamTypes.DVR].includes(streamType as any)) {
-    const liveConfig = {
-      backBufferLength: 8,
-    };
-
-    return liveConfig;
-  }
-
-  // for LL Live videos, set backBufferLenght to 4 and maxFragLookUpTolerance to 0.001
-  if ([StreamTypes.LL_LIVE, StreamTypes.LL_DVR].includes(streamType as any)) {
-    const liveConfig = {
-      backBufferLength: 4,
-      maxFragLookUpTolerance: 0.001,
-    };
-
-    return liveConfig;
-  }
+export const initialize = (
+  props: Partial<MuxMediaPropsInternal>,
+  mediaEl?: HTMLMediaElement | null,
+  core?: PlaybackCore
+) => {
+  // Automatically tear down previously initialized mux data & hls instance if it exists.
+  teardown(mediaEl, core);
 
-  return {};
-};
+  muxMediaState.set(mediaEl as HTMLMediaElement, {});
+  const nextHlsInstance = setupHls(props, mediaEl);
+  setupMux(props, mediaEl, nextHlsInstance);
+  loadMedia(props, mediaEl, nextHlsInstance);
 
-const muxMediaState: WeakMap<HTMLMediaElement, Partial<MuxMediaProps> & { error?: MediaError }> = new WeakMap();
+  const setAutoplay = setupAutoplay(props as Pick<MuxMediaProps, 'autoplay'>, mediaEl, nextHlsInstance);
+  const setPreload = setupPreload(props as Pick<MuxMediaProps, 'preload' | 'src'>, mediaEl, nextHlsInstance);
 
-export const getError = (mediaEl: HTMLMediaElement) => {
-  return muxMediaState.get(mediaEl)?.error;
+  return {
+    engine: nextHlsInstance,
+    setAutoplay,
+    setPreload,
+  };
 };
 
-export const teardown = (mediaEl?: HTMLMediaElement | null, hls?: Pick<Hls, 'detachMedia' | 'destroy'>) => {
+export const teardown = (mediaEl?: HTMLMediaElement | null, core?: PlaybackCore) => {
```

Comment:
idea(non-blocking): Per our out of band conversation, now that we're using a factory model, `teardown` can likely belong to the `PlaybackCore` object.

Reply:
I like that, any chance we might rename it to `destroy()`?
hls.js and mux-embed both have the `destroy` naming